src/libraries/System.Private.Xml.Linq/tests/XDocument.Common/BridgeHelpers.cs

```diff
@@ -1070,9 +1126,10 @@ public static void CreateBinHexTestFile(string strFileName)
             {
                 WriteToBuffer(ref WTextOnly, ref WTextOnlylen, System.BitConverter.GetBytes(strBinHex[i]));
             }
-            FilePathUtil.addStream(strFileName, new MemoryStream());
 
-            XmlWriter w = XmlWriter.Create(FilePathUtil.getStream(strFileName));
+            var ms = new MemoryStream();
+
+            XmlWriter w = XmlWriter.Create(ms);
```

Comment:
needs to stay open

Reply:
Simple test:

```csharp
using System.Xml;

{
    Console.WriteLine("StreamX.ctor");
    using var s = new StreamX();
    {
        Console.WriteLine("XmlWriter.Create");
        using XmlWriter xw = XmlWriter.Create(s);
        xw.WriteStartElement("root");
    }

    Console.WriteLine("XmlWriter.Disposed");
}

Console.WriteLine("DONE");

class StreamX : MemoryStream
{
    protected override void Dispose(bool disposing)
    {
        Console.WriteLine("StreamX.Dispose");
        base.Dispose(disposing);
    }
}
```

output:

```
StreamX.ctor
XmlWriter.Create
XmlWriter.Disposed
StreamX.Dispose
DONE
```

seems to me the XmlWriter doesn't dispose, also see the other comment

Reply:
agreed